File: Java/javaperformancetester/src/main/java/edu/imepac/javaperformancetester/models/Encoste.java
package edu.imepac.javaperformancetester.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Encoste {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToMany(fetch = FetchType.EAGER)
    private List<Vagao> vagoes = new ArrayList<>();
}

File: Kotlin/kotlinperformancetester/src/main/kotlin/edu/imepac/kotlinperformancetester/models/Encoste.kt
package edu.imepac.kotlinperformancetester.models

import jakarta.persistence.*
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
class Encoste {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private var id : Long? = null

    @ManyToMany(fetch = FetchType.EAGER)
    @JoinTable(
        name = "encoste_vagao",
        joinColumns = [JoinColumn(name = "encoste_id")],
        inverseJoinColumns = [JoinColumn(name = "vagao_id")]
    )
    val vagoes: List<Vagao> = mutableListOf()

}